extensions/cms/src/cmsResource/cmsResourceCommands.ts

```diff
@@ -95,18 +97,24 @@ export function deleteRegisteredServerCommand(appContext: AppContext, tree: CmsR
 		if (!(node instanceof RegisteredServerTreeNode)) {
 			return;
 		}
-		appContext.apiWrapper.showWarningMessage(
-			`${localize('cms.confirmDeleteServer', 'Are you sure you want to delete')} ${node.name}?`,
-			localize('cms.yes', 'Yes'),
-			localize('cms.no', 'No')).then((result) => {
-				if (result && result === localize('cms.yes', 'Yes')) {
-					appContext.cmsUtils.removeRegisteredServer(node.name, node.relativePath, node.ownerUri).then((result) => {
-						if (result) {
-							tree.notifyNodeChanged(node.parent);
-						}
-					});
+		try {
+			let result = await appContext.apiWrapper.showWarningMessage(
+				`${localize('cms.confirmDeleteServer', 'Are you sure you want to delete')} ${node.name}?`,
+				localize('cms.yes', 'Yes'),
+				localize('cms.no', 'No'));
+			if (result && result === localize('cms.yes', 'Yes')) {
+				try {
+					let removeResult = await appContext.cmsUtils.removeRegisteredServer(node.name, node.relativePath, node.ownerUri);
+					if (removeResult) {
+						tree.notifyNodeChanged(node.parent);
```

Comment:
As I mentioned in another one - I think these APIs should always return a result OR throw an error, but never return undefined. If this isn't the case let's discuss changing that so we don't have to do these kind of checks all the time (and regardless I think doing the notify is something that should happen regardless)